ngraph/core/include/ngraph/op/loop.hpp

```diff
@@ -0,0 +1,111 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <vector>
+
+#include "ngraph/factory_adapter.hpp"
+#include "ngraph/function.hpp"
+#include "ngraph/op/constant.hpp"
+#include "ngraph/op/parameter.hpp"
+#include "ngraph/op/tensor_iterator.hpp"
+#include "ngraph/op/util/sub_graph_base.hpp"
+
+namespace ngraph
+{
+    namespace op
+    {
+        namespace v5
+        {
+            /// \brief  Iterate a body over tensors, accumulating into tensors.
+            class NGRAPH_API Loop : public op::util::SubGraphOp
+            {
+            public:
+                struct SpecialBodyPorts
+                {
+                    int64_t current_iteration_input_idx =
+                        -1; // -1 means input is not provided, this input is optional
+                    int64_t body_condition_output_idx = 0; // default index, this output is required
```

Comment:
Shouldn't `body_condition_output_idx` be const value? Do we want to support scenario when `body_condition_output_idx` has value other than "0"?

Reply:
Yes, we're going to support similar behavior for both  `current_iteration_input` and `body_condition_output`.
@lazarevevgeny was mistaken a bit, everything he wrote is relevant for  `current_iteration_input` only.  Using this attribute, we will not have a disconnected Parameter in the body.
 `body_condition_output` is required for Loop op, so the attribute was only used to determine the position. It might be worth using -1 as the default and throwing an exception if no valid value was provided.